File: frontend/src/types/parking.ts
export type VehicleType = 'car' | 'motorcycle'
export type ParkingStatus = 'available' | 'occupied'

export interface ApiParkingSlot {
  id: string
  slot_number: string
  vehicle_type: VehicleType
  status: ParkingStatus
  tenant_id: string | null
  license_plate: string
  monthly_fee: number
  start_date: string | null
  note: string
  created_at: string
  updated_at: string
  tenant_first_name: string
  tenant_last_name: string
}

export interface CreateParkingSlotPayload {
  slot_number: string
  vehicle_type: VehicleType
  status: ParkingStatus
  tenant_id: string | null
  license_plate: string
  monthly_fee: number
  start_date: string | null
  note: string
}

export interface UpdateParkingSlotPayload {
  slot_number: string
  vehicle_type: VehicleType
  status: ParkingStatus
  tenant_id: string | null
  license_plate: string
  monthly_fee: number
  start_date: string | null
  note: string
}

File: backend/internal/feature/parking/domain/parking.go
package domain

import (
	"context"
	"time"
)

type VehicleType string

const (
	VehicleTypeCar        VehicleType = "car"
	VehicleTypeMotorcycle VehicleType = "motorcycle"
)

type ParkingStatus string

const (
	ParkingStatusAvailable ParkingStatus = "available"
	ParkingStatusOccupied  ParkingStatus = "occupied"
)

type ParkingSlot struct {
	ID           string        `json:"id"`
	DormitoryID  string        `json:"dormitory_id"`
	SlotNumber   string        `json:"slot_number"`
	VehicleType  VehicleType   `json:"vehicle_type"`
	Status       ParkingStatus `json:"status"`
	TenantID     *string       `json:"tenant_id"`
	LicensePlate string        `json:"license_plate"`
	MonthlyFee   float64       `json:"monthly_fee"`
	StartDate    *time.Time    `json:"start_date"`
	Note         string        `json:"note"`
	CreatedAt    time.Time     `json:"created_at"`
	UpdatedAt    time.Time     `json:"updated_at"`
	// joined
	TenantFirstName string `json:"tenant_first_name"`
	TenantLastName  string `json:"tenant_last_name"`
}

type CreateParkingSlotRequest struct {
	SlotNumber   string        `json:"slot_number"`
	VehicleType  VehicleType   `json:"vehicle_type"`
	Status       ParkingStatus `json:"status"`
	TenantID     *string       `json:"tenant_id"`
	LicensePlate string        `json:"license_plate"`
	MonthlyFee   float64       `json:"monthly_fee"`
	StartDate    *time.Time    `json:"start_date"`
	Note         string        `json:"note"`
}

type UpdateParkingSlotRequest struct {
	SlotNumber   string        `json:"slot_number"`
	VehicleType  VehicleType   `json:"vehicle_type"`
	Status       ParkingStatus `json:"status"`
	TenantID     *string       `json:"tenant_id"`
	LicensePlate string        `json:"license_plate"`
	MonthlyFee   float64       `json:"monthly_fee"`
	StartDate    *time.Time    `json:"start_date"`
	Note         string        `json:"note"`
}

type ParkingSlotRepository interface {
	FindByID(ctx context.Context, dormitoryID, id string) (*ParkingSlot, error)
	List(ctx context.Context, dormitoryID string, limit, offset int) ([]*ParkingSlot, error)
	Count(ctx context.Context, dormitoryID string) (int, error)
	Create(ctx context.Context, p *ParkingSlot) error
	Update(ctx context.Context, p *ParkingSlot) error
	Delete(ctx context.Context, dormitoryID, id string) error
}
